discovery-provider/src/tasks/index.py

```diff
@@ -331,45 +331,49 @@ def fetch_ipfs_metadata(
         .all()
     )
 
-    ipfs_metadata = {}
-    with concurrent.futures.ThreadPoolExecutor(max_workers=8) as executor:
-        futures = []
-        futures_map = {}
-        for cid, txhash in cids:
-            metadata_format = (
-                track_metadata_format
-                if cid_type[cid] == "track"
-                else user_metadata_format
+    cid_metadata = {}  # cid -> metadata
+
+    # first attempt - fetch all CIDs from replica set
+    try:
+        cid_metadata.update(
+            asyncio.run(
+                update_task.ipfs_client.fetch_metadata_from_gateway_endpoints(
+                    cid_metadata.keys(),
+                    cids_txhash_set,
+                    cid_to_user_id,
+                    user_to_replica_set,
+                    cid_type,
+                    should_fetch_from_replica_set=True,
+                )
             )
-            user_replica_set = None
-            user_id = cid_to_user_id[cid]
-            if user_id and user_id in user_to_replica_set:
-                user_replica_set = user_to_replica_set[
-                    user_id
-                ]  # user or track owner's replica set
-            future = executor.submit(
-                update_task.ipfs_client.get_metadata,
-                cid,
-                metadata_format,
-                user_replica_set,
+        )
+    except asyncio.TimeoutError:
+        # swallow exception on first attempt fetching from replica set
+        pass
+
+    # second attempt - fetch missing CIDs from other cnodes
+    if len(cid_metadata) != len(cids_txhash_set):
```

Comment:
Would it make sense for this section to only fetch the missing CIDs from the first attempt?

Reply:
that should be handled here https://github.com/AudiusProject/audius-protocol/blob/1d3aeda321db82bf7e8eb99212792b44383a3779/discovery-provider/src/utils/ipfs_lib.py#L118